animatplot/xanim.py

```diff
@@ -0,0 +1,61 @@
+import animatplot as amp
+import xarray as xr
+
+
+def animated_plot(da, anim_over='__guess__', x='x', y='y', plot_type='imshow', fps=10):
+    """
+    Function to plot an animated plot from an xarray DataArray.
+
+    The intention is that this would form the basis of a xr.DataArray.plot.animate() method.
+    """
+
+    if not isinstance(da, xr.DataArray):
+        raise TypeError('First argument must be an xarray DataArray object')
+
+    timeline, evolving_coord = _timeline_from_coords(da.coords, anim_over, fps)
+    t_axis = da.dims.index(evolving_coord)
+
+    if len(da.dims) is not 3:
+        raise NotImplementedError('Currently only plots 2D dataarrays')
```

Comment:
Should this say "only plots 3D dataarrays"

Reply:
Well it currently only plots 2D DataArrays which evolve over a 3rd dimension... Not sure what the clearest way to write that is.

Reply:
"Only animates 3D arrays"?